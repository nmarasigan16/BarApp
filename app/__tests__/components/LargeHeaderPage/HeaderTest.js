import 'react-native';
import React from 'react';
import Header from '../../../src/components/LargeHeaderPage/header/HeaderComponent';
import AdditionalInfo from '../../../src/components/LargeHeaderPage/header/AdditionalInfoComponent';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <Header
            name={'Legends'}
            location={{
                string_repr: '522 E Green St, Champaign, IL 61820',
                latitude: 40.1105053,
                longitude: -88.2311641,
            }}
            telephone={'2173557674'}
        >
        </Header>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders without location or telephone', () => {
    const tree = renderer.create(
        <Header
            name={'Legends'}
        >
        </Header>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders with shortcode telephone', () => {
    const tree = renderer.create(
        <Header
            name={'Legends'}
            telephone={'3557674'}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot()
});

it('renders without location', () => {
    const tree = renderer.create(
        <Header
            name={'Legends'}
            telephone={'2173557674'}
        >
        </Header>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders without phone number', () => {
    const tree = renderer.create(
        <Header
            name={'Legends'}
            location={{
                string_repr: '522 E Green St, Champaign, IL 61820',
                latitude: 40.1105053,
                longitude: -88.2311641,
            }}
        >
        </Header>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
