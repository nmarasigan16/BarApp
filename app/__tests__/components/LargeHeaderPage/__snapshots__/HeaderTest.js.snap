// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
      "minHeight": 333.5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Courier New",
        },
        Object {
          "color": "silver",
          "fontSize": 91.82266386900349,
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Legends
  </Text>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 6.67,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "darkgray",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Courier New",
              },
              Object {
                "alignSelf": "center",
                "color": "darkgray",
                "fontSize": 30.607554623001164,
                "marginLeft": 5,
              },
            ]
          }
        >
          522 E Green St, Champaign, IL 61820
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 6.67,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "darkgray",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Courier New",
              },
              Object {
                "color": "darkgray",
                "fontSize": 30.607554623001164,
                "marginLeft": 5,
              },
            ]
          }
        >
          (217) 355-7674
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders with shortcode telephone 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
      "minHeight": 333.5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Courier New",
        },
        Object {
          "color": "silver",
          "fontSize": 91.82266386900349,
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Legends
  </Text>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 6.67,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "darkgray",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Courier New",
              },
              Object {
                "color": "darkgray",
                "fontSize": 30.607554623001164,
                "marginLeft": 5,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders without location 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
      "minHeight": 333.5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Courier New",
        },
        Object {
          "color": "silver",
          "fontSize": 91.82266386900349,
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Legends
  </Text>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 6.67,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "darkgray",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Courier New",
              },
              Object {
                "color": "darkgray",
                "fontSize": 30.607554623001164,
                "marginLeft": 5,
              },
            ]
          }
        >
          (217) 355-7674
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders without location or telephone 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
      "minHeight": 333.5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Courier New",
        },
        Object {
          "color": "silver",
          "fontSize": 91.82266386900349,
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Legends
  </Text>
</View>
`;

exports[`renders without phone number 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
      "minHeight": 333.5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Courier New",
        },
        Object {
          "color": "silver",
          "fontSize": 91.82266386900349,
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Legends
  </Text>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 6.67,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "darkgray",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Courier New",
              },
              Object {
                "alignSelf": "center",
                "color": "darkgray",
                "fontSize": 30.607554623001164,
                "marginLeft": 5,
              },
            ]
          }
        >
          522 E Green St, Champaign, IL 61820
        </Text>
      </View>
    </View>
  </View>
</View>
`;
